test/tools/libtesteth/BlockChainHelper.h

```diff
@@ -137,10 +137,16 @@ class TestBlock
 class TestBlockChain
 {
 public:
-    TestBlockChain() : TestBlockChain(defaultGenesisBlock()) {}
-    TestBlockChain(TestBlock const& _genesisBlock);
-
-    void reset(TestBlock const& _genesisBlock);
+    enum class MiningType
+    {
+        Ethash,
+        NoProof,
+        Default
+    };
+    TestBlockChain() : TestBlockChain(defaultGenesisBlock(), MiningType::Default) {}
+    TestBlockChain(TestBlock const& _genesisBlock, MiningType _mining = MiningType::Default);
```

Comment:
This should be `explicit` constructor

Reply:
That contradicts your other requirenment. (not to change the unit tests)

Reply:
No, it shouldn't. You can still create it like `TestBlockChain bc(genesis);` even if it's `explicit`

Reply:
ok. I've added explicit keyword to this line. it works!